src/status_im/navigation/effects.cljs

```diff
@@ -1,29 +1,61 @@
 (ns status-im.navigation.effects
   (:require
+    [quo.foundations.colors :as colors]
     [quo.theme]
     [react-native.core :as rn]
     [react-native.navigation :as navigation]
+    [react-native.platform :as platform]
+    [status-im.contexts.shell.jump-to.constants :as shell.constants]
+    [status-im.contexts.shell.jump-to.utils :as jump-to.utils]
     [status-im.navigation.options :as options]
     [status-im.navigation.roots :as roots]
     [status-im.navigation.state :as state]
     [status-im.navigation.view :as views]
     [taoensso.timbre :as log]
     [utils.re-frame :as rf]))
 
-(defn- set-status-bar-color
-  [theme]
-  (rn/set-status-bar-style
-   (if (= theme :dark)
-     "light-content"
-     "dark-content")
-   true))
+(defn get-status-nav-color
+  [view-id]
+  (let [theme (or (get-in views/screens [view-id :options :theme])
+                  (quo.theme/get-theme))
+        [rnn-status-bar rn-status-bar]
+        (if (or (= theme :dark)
+                @state/alert-banner-shown?
+                (and (= view-id :shell-stack) (not (jump-to.utils/home-stack-open?))))
+          [:light "light-content"]
+          [:dark "dark-content"])
+        home-stack? (some #(= view-id %) shell.constants/stacks-ids)
+        ;; Home screen nav bar always dark due to bottom tabs
+        nav-bar-color (if (or home-stack?
+                              (= view-id :shell-stack)
+                              (= theme :dark))
+                        colors/neutral-100
+                        colors/white)
+        comp-id (if (or home-stack?
+                        (jump-to.utils/shell-navigation? view-id)
+                        (= view-id :shell))
+                  :shell-stack
+                  view-id)]
+    [rnn-status-bar rn-status-bar nav-bar-color comp-id]))
+
+(defn reload-status-nav-color-fx
+  [view-id]
+  (when (and (= @state/root-id :shell-stack) view-id)
+    (let [[rnn-status-bar rn-status-bar nav-bar-color comp-id] (get-status-nav-color view-id)]
+      (if platform/ios?
+        (rn/set-status-bar-style rn-status-bar true)
```

Comment:
this [doesn't work](https://github.com/status-im/status-mobile/issues/19072#issuecomment-1975156169) for android